### PR TITLE
[BEAM-506] Fill in the documentation/runners/flink portion of the website

### PR DESCRIPTION
**Pull request by aljoscha**
The outline of this is based on the doc for the Cloud Dataflow runner.

R: @davorbonaci and @mxm

### CODE REVIEW

**Comment by aljoscha** [comment]
@francesperry or @dhalperi, could you maybe have a look? I would like to have this in before the Beam tutorial next week in Singapore.

**Comment by mxm** [comment]
Thanks @aljoscha. Looks good! Please go ahead and merge this. We can still improve in a follow-up.

File: src/documentation/runners/flink.md [hunk 1]
```diff
@@ -6,4 +6,119 @@ redirect_from: /learn/runners/flink/
...
+</tr>
+</table>
+
+See the reference documentation for the  <span class="language-java">[FlinkPipelineOptions]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/runners/flink/FlinkPipelineOptions.html)</span><span class="language-python">[PipelineOptions](https://github.com/apache/incubator-beam/blob/python-sdk/sdks/python/apache_beam/utils/options.py)</span> interface (and its subinterfaces) for the complete list of pipeline configuration options.
+
+## Additional information and caveats
+
+### Monitoring your job
+
+You can monitor a running Flink job using the Flink JobManager Dashboard. By default, this is available at port `8081` of the JobManager node. If you have a Flink installation on your local machine that would be [http://localhost:8081](http://localhost:8081).
```
Comment (dhalperi) [inline src/documentation/runners/flink.md]: By making this a HTTP link, the website "dead link detection" is failing. We can either add `localhost` to the blacklist for dead link detection, or simply make it a code block instead.

I probably lean towards `code block`. Thoughts?

File: src/documentation/runners/flink.md [hunk 2]
```diff
@@ -6,4 +6,119 @@ redirect_from: /learn/runners/flink/
...
+$ cp target/word-count-beam-bundled-0.1.jar /path/to/flink/lib/
+```
+The Beam Quickstart Maven project is setup to use the Maven Shade plugin to create a fat jar and the `-Pflink-runner` argument makes sure to include the dependency on the Flink Runner.
+
+For actually running the pipeline you would use the Flink run script like this:
+```
+$ bin/flink run -c org.apache.beam.examples.WordCount lib/word-count-beam-0.1.jar  \
+    --inputFile=/path/to/quickstart/pom.xml  \
+    --output=/tmp/counts \
+    --runner=org.apache.beam.runners.flink.FlinkRunner
```
Comment (dhalperi) [inline src/documentation/runners/flink.md]: is the `o.a.b.r.f` needed here? In my demos recently, it has not been needed.

File: src/documentation/runners/flink.md [hunk 3]
```diff
@@ -6,4 +6,119 @@ redirect_from: /learn/runners/flink/
...
+<table class="table table-bordered">
+<tr>
+  <th>Field</th>
+  <th>Description</th>
+  <th>Default Value</th>
+</tr>
+<tr>
+  <td><code>runner</code></td>
+  <td>The pipeline runner to use. This option allows you to determine the pipeline runner at runtime.</td>
+  <td>Set to <code>org.apache.beam.runners.flink.FlinkRunner</code> to run using Flink.</td>
```
Comment (dhalperi) [inline src/documentation/runners/flink.md]: (another `o.a.b.r.f.`)

File: src/documentation/runners/flink.md (hunk 1 above)
Comment (aljoscha) [inline src/documentation/runners/flink.md]: Good catch! Changing to code block.

File: src/documentation/runners/flink.md (hunk 2 above)
Comment (aljoscha) [inline src/documentation/runners/flink.md]: In my recent tests it seemed to be necessary when using `bin/flink` to run but you're right that ideally it should not be necessary. (When running as described in the Quickstart, with the Flink local mode the `o.a.b.r.f` is not necessary.)

Did you run by executing the jar (or via `mvn exec`) or with `bin/flink`?

Comment (dhalperi) [inline src/documentation/runners/flink.md]: I ran both ways:

```
mvn compile exec:java -Dexec.mainClass=org.apache.beam.examples.WordCount \
     -Dexec.args="--inputFile=pom.xml --output=flink/counts --runner=FlinkRunner" -Pflink-runner
```

and

```
../flink-1.1.2/bin/flink run target/word-count-beam-0.1.jar --inputFile=pom.xml --output=flink_cluster/counts --runner=FlinkRunner
```

I am working on making the changes I made into Beam (https://github.com/apache/incubator-beam/pull/1465) but here's the `flink-cluster` profile:

```xml
    <profile>
      <id>flink-cluster</id>
      <!-- Produces a Flink fat jar to run WordCount. -->
  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-shade-plugin</artifactId>
        <version>2.4.1</version>
        <executions>
          <execution>
            <phase>package</phase>
            <goals>
              <goal>shade</goal>
            </goals>
            <configuration>
              <transformers>
                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                  <mainClass>org.apache.beam.examples.WordCount</mainClass>
                </transformer>
              </transformers>
              <filters>
                <filter>
                  <artifact>*:*</artifact>
                  <excludes>
                    <exclude>META-INF/LICENSE</exclude>
                  </excludes>
                </filter>
              </filters>
            </configuration>
          </execution>
        </executions>                                                                               
      </plugin>                                                                                     
                                                                                                    
      <plugin>                                                                                      
        <groupId>org.apache.maven.plugins</groupId>                                                 
        <artifactId>maven-compiler-plugin</artifactId>                                              
        <configuration>                                                                             
          <source>1.7</source>                                                                      
          <target>1.7</target>                                                                      
        </configuration>                                                                            
      </plugin>                                                                                     
                                                                                                    
    </plugins>                                                                                      
  </build>                                                                                          
                                                                                                    
      <dependencies>                                                                                
        <dependency>                                                                                
          <groupId>org.apache.beam</groupId>                                                        
          <artifactId>beam-runners-flink_2.10</artifactId>                                          
          <version>0.4.0-incubating-SNAPSHOT</version>                                              
          <scope>runtime</scope>                                                                    
        </dependency>                                                                               
      </dependencies>                                                                               
    </profile>                                                                                      
```

Comment (dhalperi) [inline src/documentation/runners/flink.md]: in retrospect, I ought to be able to put this in the `flink-runner` profile -- no need to have two profiles :)

Comment (aljoscha) [inline src/documentation/runners/flink.md]: Thanks! You're absolutely right, this works very smoothly, even without using `bin/flink`. Last time I checked was with 0.3.0, now I checked with 0.4.0-SNAPSHOT and the correct Flink version.

I'll quickly update the doc. In a separate PR I'll also update the Quickstart. My instructions there are correct but more complex than needed, as I just found out.

**Comment by aljoscha** [comment]
@dhalperi I simplified the instructions based on your feedback, PTAL.

**Comment by dhalperi** [comment]
Looks great. Thanks Aljoscha!

(I didn't know whether you were going to merge or wanted reviewer to, so I went ahead and did it. For future ref: do you have the website jekyll stuff set up?)

**Comment by davorbonaci** [comment]
Thanks @aljoscha, @mxm and @dhalperi -- sorry for my delays here.

**Comment by aljoscha** [comment]
Thanks for the review! (and merging)

@dhalperi I have the Jekyll setup, yes. I think I'm the only one rebuilding the website from outside the west coast, though, and I'm always manually rolling back timestamp changes in the generated files. 😃

**Pull request closed without merging**